File: python/sparklabs/batch/sales/dataframe/spark_salting.py
from pyspark.sql import functions as F

from sparklabs.batch.sales.dataframe import utils


def run() -> None:
    (
        utils.spark.range(20)
        .withColumn("salt", (F.rand() * 10).cast("int"))
        .show(truncate=False)
    )


if __name__ == "__main__":
    run()

File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/dataframe/SparkSalting.scala
package io.devorbit.sparklabs.batch.sales.dataframe

import org.apache.spark.sql.expressions.Window
import org.apache.spark.sql.functions._

object SparkSalting extends App {

  val df2 =
    Utils.sparkSession.range(20)
      .withColumn("salt", (rand() * 10).cast("int"))
      .show(false)

//  Thread.sleep(1000000)
}
